### DISCUSSION TITLE
coverage issue

### DISCUSSION THREAD

**Discussion by Cyan-123** [cvc5/cvc5#7250] [Q&A] [answered]
Here are two questions:
1. After compiling with ". /configue --coverage" , when using cvc4 to test the smt2 file, the running result is normal but the corresponding gcda file will not be generated or updated, and the gcda file will be generated only when the error is reported.

2. When testing regression cases with ctest, adding run parameters is not valid. For example, adding an A.smt2 file of QF_S logic to regress4, "; COMMAND-LINE:-- strings-exp" is set in the A.smt2 file, but when running "ctest -R regress4", "--string-exp" has no effect.
![image](https://user-images.githubusercontent.com/54757112/124273038-19c49380-db72-11eb-8924-cd2de9dcc19c.png)

**Comment by mpreiner** [comment, marked as answer]
What steps did you perform after configuring with `--coverage`?

If you want to run only specific files and generate the coverage report you should use the `coverage-reset` and `coverage` targets as described here: https://github.com/cvc5/cvc5/wiki/Developer-Guide#code-coverage-reports
